lib/events/setter.go

```diff
@@ -135,6 +139,14 @@ func (c *Preparer) PrepareSessionEvent(event apievents.AuditEvent) (apievents.Pr
 	return preparedEvent, nil
 }
 
+func (c *Preparer) nextIndex() int64 {
+	if !c.cfg.StartTime.IsZero() {
```

Comment:
Is it possible for all items have a start time, so we don't need to keep the fallback logic?